core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/FillVector.java

```diff
@@ -0,0 +1,347 @@
+        int size = pageSize * DataTypes.SHORT_INT.getSizeInBytes();
+        for (int i = 0; i < size; i += DataTypes.SHORT_INT.getSizeInBytes()) {
+          vector.putFloat(rowId++, (ByteUtil.valueOf3Bytes(pageData, pageIndex + i) / floatFactor));
+        }
+        pageIndex += size;
+      } else if (pageDataType == DataTypes.INT) {
+        int size = pageSize * DataTypes.INT.getSizeInBytes();
+        for (int i = 0; i < size; i += DataTypes.INT.getSizeInBytes()) {
+          vector.putFloat(rowId++,
+              (ByteUtil.toIntLittleEndian(pageData, pageIndex + i) / floatFactor));
+        }
+        pageIndex += size;
+      } else if (pageDataType == DataTypes.LONG) {
+        int size = pageSize * DataTypes.LONG.getSizeInBytes();
+        for (int i = 0; i < size; i += DataTypes.LONG.getSizeInBytes()) {
+          vector.putFloat(rowId++,
+              (float) (ByteUtil.toLongLittleEndian(pageData, pageIndex + i) / factor));
+        }
+        pageIndex += size;
+      } else {
+        throw new RuntimeException("internal error: " + this.toString());
+      }
+    } else if (vectorDataType == DataTypes.DOUBLE && factor != 0.0) {
+      if (pageDataType == DataTypes.BOOLEAN || pageDataType == DataTypes.BYTE) {
+        for (int i = 0; i < pageSize; i++) {
+          vector.putDouble(i, (pageData[pageIndex++] / factor));
+        }
+      } else if (pageDataType == DataTypes.SHORT) {
+        int size = pageSize * DataTypes.SHORT.getSizeInBytes();
+        for (int i = 0; i < size; i += DataTypes.SHORT.getSizeInBytes()) {
+          vector
+              .putDouble(rowId++, (ByteUtil.toShortLittleEndian(pageData, pageIndex + i) / factor));
+        }
+        pageIndex += size;
+      } else if (pageDataType == DataTypes.SHORT_INT) {
+        int size = pageSize * DataTypes.SHORT_INT.getSizeInBytes();
+        for (int i = 0; i < size; i += DataTypes.SHORT_INT.getSizeInBytes()) {
+          vector.putDouble(rowId++, (ByteUtil.valueOf3Bytes(pageData, pageIndex + i) / factor));
+        }
+        pageIndex += size;
+      } else if (pageDataType == DataTypes.INT) {
+        int size = pageSize * DataTypes.INT.getSizeInBytes();
+        for (int i = 0; i < size; i += DataTypes.INT.getSizeInBytes()) {
+          vector.putDouble(rowId++, (ByteUtil.toIntLittleEndian(pageData, pageIndex + i) / factor));
+        }
+        pageIndex += size;
+      } else if (pageDataType == DataTypes.LONG) {
+        int size = pageSize * DataTypes.LONG.getSizeInBytes();
+        for (int i = 0; i < size; i += DataTypes.LONG.getSizeInBytes()) {
+          vector
+              .putDouble(rowId++, (ByteUtil.toLongLittleEndian(pageData, pageIndex + i) / factor));
+        }
+        pageIndex += size;
+      } else {
+        throw new RuntimeException("Unsupported datatype : " + pageDataType);
+      }
+    } else if (pageDataType == DataTypes.BYTE_ARRAY) {
+      if (vectorDataType == DataTypes.STRING) {
```

Comment:
DirectCompressCodec & AdaptiveIntegralCodec had code related to decimal, 
I think you have removed it while extracting common code. 
This is impact primitive type functionality also. I guess no testcase present. so, you CI didn't catch.

Reply:
Also if more than one complex type children has decimal, this vector info has to be redesigned. it cannot contain all decimal column vectors now

Reply:
okay